tools/cpp/nm_validate_static_library.sh.tpl

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+#
+# Copyright 2023 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+set -eu
+
+DUPLICATE_SYMBOLS=$(
+  "%{nm}" --defined-only --demangle --print-file-name "$1" |
+  LC_ALL=C sort --key=3 |
+  uniq -D --skip-field=2)
+if [[ -n "$DUPLICATE_SYMBOLS" ]]; then
+  echo "Duplicate symbols found in $1:" | tee "$2"
+  echo "$DUPLICATE_SYMBOLS" | tee --append "$2"
+  exit 1
+else
+  touch "$2"
+fi
```

Comment:
I tried my updated branch which is based on this, and the validation failed with a lot of duplicate symbols. Not sure why there are so many duplicate symbols.

Reply:
Interestingly, I tried stripping the static library before this script did validation, and it seems to fix the issue partially. The duplicate symbols are not outputted. 

But I get things like:
```
nm: test.pic.o: no symbols
```

Reply:
Do you happen to have an open-source project that this reproduces on?

Reply:
Yes, the Drake library:
https://github.com/RobotLocomotion/drake/issues/16882

Reply:
Thanks, I now know why and will update the script to fix it.

Reply:
@aminya Could you try again with the current state of the PR? I fixed a bunch of issues and verified it on Protobuf.